benchkit/shell/shellasync.py

```diff
@@ -122,6 +123,44 @@ def wait(self, timeout: Optional[int] = None) -> None:
                 returncode=self._error_code,
             )
 
+    def find_matching_ssh(self, local_pid):
```

Comment:
I'm wondering whether we shouldn't kill the actual process instead of the ssh connection? Then ssh would take over and exits itself?

Reply:
The version before this commit kills the `ssh connection` and not the actual process.
This commit is doing what you're suggestion, but to kill the actual process we need to find it on the remote machine.

For that, we first call `find_matching_ssh` to find which ssh on the remote machine is answering/connected to the remote call in the local machine.
The actual process that we want to kill is one of the children on this "matching ssh", so we look into its subtree and kill all processes there starting from the leaves